Fix: SSH bug in windows

SuvarnaNarayanan opened the pull request:
With `SshMachine`, if we were to use windows specific path, it translates into something like `/c/Users/...` which isn't a compatible path for the `scp` call in windows. I have changed the logic that calls the translation function to run only when the underlying platform isn't windows, which fixes the problem.

Also, some basic test cases fix to prevent `ImportError` when trying to get `printenv`

philosopherstonerush committed:
ssh win fix + test import env case fix

henryiii commented on tests/test_env.py:
```diff
@@ -5,7 +5,7 @@
 from plumbum._testtools import skip_on_windows
 
-with contextlib.suppress(ModuleNotFoundError):
+with contextlib.suppress(ModuleNotFoundError, ImportError):
```
```suggestion
with contextlib.suppress(ImportError):
```

`ModuleNotFoundError` is an `ImportError`

Not sure about the other change yet, just noting this quickly.

henryiii committed:
fix: detect system

Assisted-by: ClaudeCode:claude-opus-4.8
Signed-off-by: Henry Schreiner <henryfs@princeton.edu>

henryiii committed:
fix: protect against potential issues raising instead of returning False

Assisted-by: OpenCode:Kimi-K2.6
Signed-off-by: Henry Schreiner <henryfs@princeton.edu>

henryiii committed:
fix: only drive-letter translate the local scp path

download/upload applied _translate_drive_letter to both src and dst, but
only one side is local. The translation exists so the local cygwin/MSYS
scp binary can parse native c:\... paths; the remote path follows the
remote's conventions and must be passed through verbatim. Translating the
remote side silently corrupted remote POSIX paths containing a colon (a
legal filename char) on Windows hosts using cygwin/MSYS scp.

Now download translates only dst and upload only src. Adds regression
tests asserting which argument is translated.

Assisted-by: ClaudeCode:claude-opus-4.8
